### PR TITLE
Flow test --ui supports streaming (for python chat flow)

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py [hunk 1]
```diff
@@ -262,6 +262,23 @@ def py_file(self):
...
+                    lambda key: self.executable.outputs[key].is_chat_output,
+                    self.executable.outputs.keys(),
+                )
+            )
+        except StopIteration:
+            output_name = None
+        return output_name
+
+    @property
+    def stream(self):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py:279]: stream -> is_streaming

File: src/promptflow/promptflow/_sdk/data/executable/main.py [hunk 2]
```diff
@@ -13,6 +15,43 @@
...
+            if hasattr(output.gi_frame.f_locals, "proxy"):
+                proxy = output.gi_frame.f_locals["proxy"]
+                generator_record[output] = proxy
+            else:
+                generator_record[output] = list(output)
+                output = generator_record[output]
+    return output
+
+
+def resolve_generator(flow_result):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/data/executable/main.py:38]: looks like we dulplicate the code from cli? could we reuse that code?

File: src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py [hunk 3]
```diff
@@ -262,6 +262,23 @@ def py_file(self):
     def flow_path(self):
         return self.flow_dag_path.as_posix()
 
+    @property
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py:265]: can demo this to the team in next meeting

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/data/executable/utils.py [hunk 4]
```diff
@@ -58,24 +58,24 @@ def item_render_message(value, key=None):
...
+            st.markdown(f"{json_dumps(value)},")
         else:
-            st.markdown(f"`{key}: {json_dumps(value)},`")
+            st.markdown(f"{key}: {json_dumps(value)},")
 
 
 def list_iter_render_message(message_items):
     if is_list_contains_rich_text(message_items):
-        st.markdown("`[ `")
+        st.markdown("[ ")
```
Comment (D-W-) [inline src/promptflow/promptflow/_sdk/data/executable/utils.py:68]: Why changed this?

**Review by D-W- — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/data/executable/utils.py (hunk 4 above)
Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/data/executable/utils.py:68]: if add `` , streamlit will regard contents as code block and show as code block format.

**Review by D-W- — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**